paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/ManageScreenInteractor.kt

```diff
@@ -59,12 +65,20 @@ internal class DefaultManageScreenInteractor(
         val displayablePaymentMethods = paymentMethods?.map {
             it.toDisplayableSavedPaymentMethod(providePaymentMethodName, paymentMethodMetadata)
         } ?: emptyList()
+
+        if (noPaymentMethodsAvailableToManage(displayablePaymentMethods)) {
```

Comment:
This doesn't seem like the right location to do this.

I think we want to have a separate `collect` in the constructor. This will require some work, but we do something similar in `ModifiableEditPaymentMethodViewInteractor`.

The reason for this is because we should separate our reading and mutating of state.

Reply:
Updated. Manually verified that everything still works properly + interactor tests still pass.